src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,7 +35,20 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+
+        
+        String[] splitAddress = address.split("\\, ");
```

Comment:
splitAddress might not have a length of 4, depending on the user input. This would cause Array index out of bound exceptions when creating the individual object later on in line 48 to 51. Though this could of course be solved in an alternative way, by putting a cascading switch statement, i.e.

int length = splitAddress.length;

switch (length){
    case 4: postalCode = new ...;
    case 3: unit = new ....;
    case 2: street = new ...;
    case 1: block = new ....; break;
}

Of course this assumes that the user has to enter the address components in the correct order, and cannot specify a component without first specifying the previous one, e.g. user cannot type in postal code without having typed in a unit number. 

A way to solve this might be to split the string and then perform checks similar to the checks used split the entire user input into name and email etc. We might then be able to correctly assign a postal code input as a postal code, even if the use only enters the block number, street name, and postal code.